cunumeric/module.py

```diff
@@ -209,44 +205,30 @@ def choose(a, choices, out=None, mode="raise"):
     return array.choose(choices=choices, out=out, mode=mode, stacklevel=2)
 
 
+def diag_helper(a, offset=0, axes=None, extract=True):
```

Comment:
I think we could unify module/array:diagonal and module/array:diag_helper, doing something like the following:

module.py:
```
def diagonal(a, offset=0, axis1=None, axis2=None, extract=True, axes=None):
    array = ndarray.convert_to_cunumeric_ndarray(a)
    return array.diagonal(offset=offset, axis1=axis1, axis2=axis2, extract=extract, axes=axes)
```

array.py:
```
def diagonal(a, offset=0, axis1=None, axis2=None, extract=True, axes=None):
    if type(axis1) == int and type(axis2) == int:
        if axes is not None:
            raise ValueError("Either axis1/axis2 or axes must be supplied")
        axes = (axis1, axis2)
    elif axis1 is not None or axis2 is not None:
        raise ValueError("Either axis1/axis2 or axes must be supplied")
    ...
```

Reply:
Sure, but do we  want to expose `axes` to general users of `cunumeric`? This will be a deviation from `numpy`. I am asking mostly to understand how closely we need to follow `numpy` interface.

Reply:
We are free to extend the NumPy APIs as we see fit. If we don't support something that NumPy does then we need to document that, or throw an informative exception.

Reply:
OK, understood.